<?xml version="1.0"?>
<launch>
    <!-- Launch file arguments -->

    <arg name="baud_rate" default="250000"/>

    <!-- Launch the dynamic serial manager node -->
    <node name="rosserial_dynamic" pkg="svea_examples" type="rosserial_dynamic.py" output="screen">
        <!-- Set parameters for the node -->
        <param name="baud" value="$(arg baud_rate)"/>
        <param name="scan_rate" value="1"/>
    </node>
    
    <!-- Debugging options -->
    <arg name="enable_debug" default="false"/>
    <group if="$(arg enable_debug)">
        <node name="rqt_console" pkg="rqt_console" type="rqt_console" output="screen"/>
        <node name="rqt_logger_level" pkg="rqt_logger_level" type="rqt_logger_level" output="screen"/>
    </group>
</launch>
